### ISSUE TITLE
[RELEASE] Data Prepper 1.2.1 tar.gz artifacts

### ISSUE THREAD

**Issue by dlvenable** [opensearch-project/opensearch-build#1252]
### Is your feature request related to a problem? Please describe

The Data Prepper team plans to release Data Prepper as tar.gz files. There are four tar.gz files for distribution:

* Linux
* Linux with JDK
* ~macOS~ (deferred)
* ~macOS with JDK~ (deferred)

Planned release date: TBD; mid-Dec 2021

### Describe the solution you'd like

The Data Prepper contributors would like to build these artifacts and upload them to an S3 bucket. From there, we'd like the build team to copy and sign these artifacts with their existing signing infrastructure. Then the build team can make them available as public links.

Once the links are ready, the Data Prepper team can update the OpenSearch.org download documentation page to include them.

### Additional context

The Data Prepper work necessary for this change is:
https://github.com/opensearch-project/data-prepper/issues/696

**Comment by dlvenable** [comment]
This issue is requesting tar.gz files for the planned Docker release which is tracked in #1106.

**Comment by zelinh** [comment]
[Triage] Data prepper team will need to attend infra's office hour and work with infra's oncall of the week to release tarball. They have changed the release date so this is not urgent.

**Comment by gaiksaya** [comment]
Hi @dlvenable , @elfisher and @stockholmux ,
Do we know what should be the downloading URL for data-prepper?

Suggestion (following what we have for logstash on our website's download page):

```
https://artifacts.opensearch.org/data-prepper/opensearch-data-prepper-1.2.1-linux-x64.tar.gz

https://artifacts.opensearch.org/data-prepper/opensearch-data-prepper-1.2.1-linux-x64.tar.gz.sig

https://artifacts.opensearch.org/data-prepper/opensearch-data-prepper-jdk-1.2.1-linux-x64.tar.gz

https://artifacts.opensearch.org/data-prepper/opensearch-data-prepper-jdk-1.2.1-linux-x64.tar.gz.sig
```

**Comment by dlvenable** [comment]
This is the current OpenSearch URL:

```
https://artifacts.opensearch.org/releases/bundle/opensearch/1.2.3/opensearch-1.2.3-linux-x64.tar.gz
```

Based on that, I'd think:

```
https://artifacts.opensearch.org/releases/bundle/data-prepper/1.2.1/opensearch-data-prepper-1.2.1-linux-x64.tar.gz
```

The path prefix is something I'm flexible on.

**Comment by gaiksaya** [comment]
bundle is the opensearch-bundle so that is specific to bundles that we create. Version prefix sounds good too just that we do not have that for other stand-alone components. Also version number is present in the artifact name as well.
@peternied @peterzhuamazon  what do you think?

**Comment by dlvenable** [comment]
I think having the version number in the path can help, especially if the server supports (or were to support listing). So I'd like to keep it in the path.

So to summarize, I'm looking to have it at a path which ends in:

```
data-prepper/1.2.1/opensearch-data-prepper-1.2.1-linux-x64.tar.gz
```

I am agnostic to any path prefix (or lack thereof) and leave that whatever works best for your team.

**Comment by gaiksaya** [comment]
> I think having the version number in the path can help, especially if the server supports (or were to support listing). So I'd like to keep it in the path.
> 
> So to summarize, I'm looking to have it at a path which ends in:
> 
> ```
> data-prepper/1.2.1/opensearch-data-prepper-1.2.1-linux-x64.tar.gz
> ```
> 
> I am agnostic to any path prefix (or lack thereof) and leave that whatever works best for your team.

Sounds good. @elfisher @stockholmux any objections?

**Comment by gaiksaya** [comment]
Hi @dlvenable ,

Artifacts have been promoted to prod. Do you have an update to update our website? 
Looks like all we have is a docker image today and we need to add option for tarball and respective signatures.
Thanks!

**Comment by dlvenable** [comment]
@gaiksaya , Yes, I am working a PR presently for the project-website project.

I will also verify the artifacts from the build.

**Comment by dlvenable** [comment]
@gaiksaya , I confirmed that the artifacts work as expected.

I also pushed a PR for the project-website: https://github.com/opensearch-project/project-website/pull/622

**Comment by gaiksaya** [comment]
Thank you! We can close this issue once the PR is merged.

**Comment by stockholmux** [comment]
opensearch-project/project-website#622 is merged.
